packages/storybook/src/ui/molecules/table/table.test.tsx

```diff
@@ -60,3 +69,261 @@ describe('Component overrride', () => {
     expect(screen.getByRole('button', { name: 'This cell is customised' })).toBeVisible()
   })
 })
+
+describe('Table with infinite scroll pagination (x)', () => {
+  it('navigates to next page using the button when scroll is not available', () => {
+    render(
+      <div style={{ maxHeight: '400px' }}>
+        <Table tableData={mockTableDataForPagination} />
+      </div>
+    )
+
+    expect(screen.getAllByRole('row').length).toEqual(11)
+
+    userEvent.click(screen.getByRole('button'))
+
+    expect(screen.getAllByRole('row').length).toEqual(21)
+  })
+
+  it('navigates to next page using the scroll when scroll is available', () => {
+    render(
+      <div style={{ maxHeight: '200px' }}>
+        <Table tableData={mockTableDataForPagination} />
+      </div>
+    )
+
+    expect(screen.getAllByRole('row').length).toEqual(11)
+
+    fireEvent.scroll(screen.getByRole('table').parentElement as HTMLElement, { target: { scrollY: 100 } })
+
+    expect(screen.getAllByRole('row').length).toEqual(21)
+
+    fireEvent.scroll(screen.getByRole('table').parentElement as HTMLElement, { target: { scrollY: 100 } })
+
+    expect(screen.getAllByRole('row').length).toEqual(31)
+  })
+})
+
+describe('Table with infinite scroll pagination (y)', () => {
+  it('navigates to next page using the button when scroll is not available', () => {
+    render(
+      <div style={{ maxWidth: '100%' }}>
+        <Table tableData={mockTableDataForVerticalPagination} headerAxis="y" />
+      </div>
+    )
+
+    expect(screen.getAllByRole('columnheader').length).toEqual(11)
+
+    userEvent.click(screen.getByRole('button'))
+
+    expect(screen.getAllByRole('columnheader').length).toEqual(21)
+  })
+
+  it('navigates to next page using the scroll when scroll is available', () => {
+    render(
+      <div style={{ maxWidth: '1050px' }}>
+        <Table tableData={mockTableDataForVerticalPagination} headerAxis="y" />
+      </div>
+    )
+
+    expect(screen.getAllByRole('columnheader').length).toEqual(11)
+
+    fireEvent.scroll(screen.getByRole('table').parentElement as HTMLElement, { target: { scrollX: 100 } })
+
+    expect(screen.getAllByRole('columnheader').length).toEqual(21)
+
+    fireEvent.scroll(screen.getByRole('table').parentElement as HTMLElement, { target: { scrollX: 100 } })
+
+    expect(screen.getAllByRole('columnheader').length).toEqual(31)
+  })
+})
+
+describe('Table with infinite scroll pagination (x) [MANUAL]', () => {
+  it('navigates to next page using the button when scroll is not available', async () => {
```

Comment:
I think it'd be cleaner and better practice to test the hook and the component separately - we're testing a few too many things at once here. 
To test the component in isolation you can mock out the hook using `jest.spyOn`. 
https://jestjs.io/docs/mock-function-api
(if the docs aren't clear enough there are some examples in the ehr-client repo, specifically for the timeline and tabular summary widgets)